Set-DomainUser.ps1

```diff
@@ -1,720 +1,1124 @@
+        
+        Copy-Item -path "$psmRootInstallLocation\Hardening\PSMHardening.ps1" -Destination $PSMHardeningBackupFileName
+        Copy-Item -path "$psmRootInstallLocation\Hardening\PSMConfigureAppLocker.ps1" -Destination $PSMConfigureAppLockerBackupFileName
+        Copy-Item -Path "$psmRootInstallLocation\basic_psm.ini" -Destination $BasicPSMBackupFileName
+        
+        If (!(Test-Path $PSMHardeningBackupFileName)) {
+            Write-Error "Failed to backup PSMHardening.ps1" -ErrorAction Stop
+        }
+        ElseIf (!(Test-Path $PSMConfigureAppLockerBackupFileName)) {
+            Write-Error "Failed to backup PSMConfigureAppLocker.ps1" -ErrorAction Stop
+        }
+        ElseIf (!(Test-Path $BasicPSMBackupFileName )) {
+            Write-Error "Failed to backup basic_psm.ini" -ErrorAction Stop
+        }
+    }
+    catch {
+        write-output "Could not copy one of the scripts to backup. Exiting"
+        write-output $_
+        exit
+    }
+}
+
+Function Update-PSMConfig {
+    <#
+    .SYNOPSIS
+    Updates PSM scripts and basic_psm.ini to have domain user(s) in them rather than local user(s).
+    .DESCRIPTION
+    Updates PSM scripts and basic_psm.ini to have domain user(s) in them rather than local user(s).
+    .PARAMETER psmRootInstallLocation
+    PSM root installation folder
+    .PARAMETER domain
+    The user domain
+    .PARAMETER PsmConnectUsername
+    PSM Connect User name
+    .PARAMETER PsmAdminUsername
+    PSM Admin Connect user name
+    #>
+    param (
+        [Parameter(Mandatory = $true)]
+        $psmRootInstallLocation,
+        [Parameter(Mandatory = $true)]
+        $domain,
+        [Parameter(Mandatory = $true)]
+        $PsmConnectUsername,
+        [Parameter(Mandatory = $true)]
+        $PsmAdminUsername
+    )
+    try {
+        #PSMHardening
+        #-------------------------   
+        $psmHardeningContent = Get-Content -Path $psmRootInstallLocation\Hardening\PSMHardening.ps1
+        
+        $newPsmHardeningContent = $psmHardeningContent -replace '^(\$PSM_CONNECT_USER\s*=) .*', ('$1 "{0}\{1}"' -f $domain, $PsmConnectUsername)
+        $newPsmHardeningContent = $newPsmHardeningContent -replace '^(\$PSM_ADMIN_CONNECT_USER\s*=) .*$', ('$1 "{0}\{1}"' -f $domain, $PsmAdminUsername)
+        $newPsmHardeningContent | Set-Content -Path "$psmRootInstallLocation\Hardening\test-psmhardening.ps1"
+        
+        #PSMApplocker    
+        #------------------------- 
+        
+        
+        $psmApplockerContent = Get-Content -Path $psmRootInstallLocation\Hardening\PSMConfigureApplocker.ps1
+        
+        $newPsmApplockerContent = $psmApplockerContent -replace '^(\$PSM_CONNECT\s*=) .*', ('$1 "{0}\{1}"' -f $domain, $PsmConnectUsername)
+        $newPsmApplockerContent = $newPsmApplockerContent -replace '^(\$PSM_ADMIN_CONNECT\s*=) .*$', ('$1 "{0}\{1}"' -f $domain, $PsmAdminUsername)
```

Comment:
Changed how scripts are edited so that subsequent runs will modify them successfully